src/js/utils/parser.ts

```diff
@@ -81,21 +81,22 @@ export function serialize(val) {
 }
 
 // Returns the integer value a of css string (e.g. '420px')
-export function parseDimension(dimension) {
-    if (typeof dimension === 'string') {
-        if (dimension === '') {
-            return 0;
-        } else if (dimension.lastIndexOf('%') > -1) {
-            return dimension;
-        }
-        return parseInt(dimension.replace('px', ''), 10);
+export function parseDimension(dimension: string): number {
+    if (isValidNumber(dimension)) {
+        return parseInt(dimension);
+    }
+
+    if (dimension === '') {
+        return 0;
+    } if (dimension.lastIndexOf('%') > -1) {
+        return parseInt(dimension, 10);
     }
-    return dimension;
+    return parseInt(dimension.replace('px', ''), 10);
 }
 
 // Returns a formatted time string from "mm:ss" to "hh:mm:ss" for the given number of seconds
-export function timeFormat(sec, allowNegative) {
-    if ((sec <= 0 && !allowNegative) || isNaN(parseInt(sec))) {
+export function timeFormat(sec: number, allowNegative?: boolean): string {
+    if (isNaN(sec) || !isFinite(sec) || (sec <= 0 && !allowNegative)) {
```

Comment:
are we positive `sec` is only a string when we want to throw it out? previously we would try to parse int before getting rid of iit

Reply:
Somehow I glanced over that. Good catch!